src/org/jetbrains/plugins/scala/codeInspection/collections/EmulateFlattenInspection.scala

```diff
@@ -0,0 +1,39 @@
+package org.jetbrains.plugins.scala.codeInspection.collections
+
+import org.jetbrains.plugins.scala.codeInspection.InspectionBundle
+import org.jetbrains.plugins.scala.extensions.ResolvesTo
+import org.jetbrains.plugins.scala.lang.psi.api.expr._
+
+/**
+  * @author Lukasz Piepiora
+  */
+class EmulateFlattenInspection extends OperationOnCollectionInspection {
+  override def possibleSimplificationTypes: Array[SimplificationType] =
+    Array(FlattenSimplification)
+}
+
+object FlattenSimplification extends SimplificationType {
+
+  override def hint: String = InspectionBundle.message("replace.with.flatten")
+
+  override def getSimplification(expr: ScExpression): Option[Simplification] =
+    expr match {
+      case seqOfSeqs `.flatMap` (identityOperation()) =>
+        Some(replace(expr).withText(invocationText(seqOfSeqs, "flatten")))
+      case _ => None
+    }
+
+  private object identityOperation {
+    def unapply(expr: ScExpression): Boolean = stripped(expr) match {
```

Comment:
I would use syntax for infix patterns here: `_'.identity' (underscore())` , but it's up to you.